JustSaying/AwsTools/MessageHandling/SnsTopicByName.cs

```diff
@@ -96,5 +97,72 @@ public async Task<bool> CreateAsync()
 
             return false;
         }
+
+        public async Task<bool> CreateWithEncryptionAsync(ServerSideEncryption config)
+        {
+            var created = await CreateAsync().ConfigureAwait(false);
+
+            ServerSideEncryption = await ExtractServerSideEncryptionFromTopicAttributes().ConfigureAwait(false);
+
+            await EnsureServerSideEncryptionIsUpdatedAsync(config).ConfigureAwait(false);
+
+            return created;
+        }
+
+        private async Task<ServerSideEncryption> ExtractServerSideEncryptionFromTopicAttributes()
+        {
+            var attributesResponse = await Client.GetTopicAttributesAsync(Arn).ConfigureAwait(false);
+            
+            if (!attributesResponse.Attributes.TryGetValue(JustSayingConstants.AttributeEncryptionKeyId, out var encryptionKeyId))
+            {
+                return null;
+            }
+
+            return new ServerSideEncryption
+            {
+                KmsMasterKeyId = encryptionKeyId
+            };
+        }
+
+        private async Task EnsureServerSideEncryptionIsUpdatedAsync(ServerSideEncryption config)
+        {
+            if (ServerSideEncryptionNeedsUpdating(config))
+            {
+                var request = new SetTopicAttributesRequest
+                {
+                    TopicArn = Arn,
+                    AttributeName = JustSayingConstants.AttributeEncryptionKeyId,
+                    AttributeValue = config?.KmsMasterKeyId ?? string.Empty
+                };
+
+                var response = await Client.SetTopicAttributesAsync(request).ConfigureAwait(false);
+
+                if (response.HttpStatusCode == HttpStatusCode.OK)
+                {
+                    ServerSideEncryption = string.IsNullOrEmpty(config?.KmsMasterKeyId)
+                        ? null
+                        : config;
+                }
+                else
+                {
+                    _log.LogWarning($"Request to set topic attribute {request.AttributeName} to {request.AttributeValue} failed with status code {response.HttpStatusCode}.");
```

Comment:
This would be better if it used structured logging syntax:

```csharp
_log.LogWarning("Request to set topic attribute {TopicAttributeName} to {TopicAttributeValue} failed with status code {HttpStatusCode}.", request.AttributeName, request.AttributeValue, response.HttpStatusCode);
```

Reply:
Done :)